daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala

```diff
@@ -1078,10 +1060,9 @@ object JsonLedgerClient {
         "templateId" -> args.templateId.toJson,
         "key" -> LfValueCodec.apiValueToJsValue(args.key))
     implicit val fetchReader: RootJsonReader[FetchResponse] = v =>
-      v.asJsObject.getFields("result") match {
-        case Seq(JsNull) => FetchResponse(None)
-        case Seq(r) => FetchResponse(Some(r.convertTo[ActiveContract]))
-        case _ => deserializationError(s"Could not parse FetchResponse: $v")
+      v match {
+        case JsNull => FetchResponse(None)
+        case r => FetchResponse(Some(r.convertTo[ActiveContract]))
     }
```

Comment:
```suggestion
      FetchResponse(v.convertTo[Option[ActiveContract]])
```

Reply:
annoyingly, spray only provides a JsonFormat for Option which means we also need a JsonFormat (and thereby a writer for) ActiveContrct. I’ve just written my own reader-only implicit for Option now.